### ISSUE TITLE
renderBootstrap3 BootstrapBasicForm renders hiddenField as a normal field

### ISSUE THREAD

**Issue by ricksanchez** [yesodweb/yesod#1438]
It is rendering the whole row as it was visible:
`<div class="form-group required "><label for="hident4"></label><input type="hidden" id="hident4" name="f3" value="mysecret"></div>`
this results in ugly layouts

**Comment by felixSchl** [comment]
Experiencing the same issue, how are we supposed to work around this? It seems the `FieldView` does not even carry enough information to make a decision on how to render it at render time

**Comment by ricksanchez** [comment]
This is related to the fact that the applicative form is only for the most elementary forms in the most generic way.
From the basic level to advanced, it is better to use the monadic.

I would recommend you to use the monadic instead, and make a bunch of functions to match your particular user case. That is what I am doing since then. Any realistic post 90s form (with placeholders, tooltips, css classes, id for jQuery enhancements, interrelated widgets with add/remove, help...) will require a similar approach.

Anyway, I think renderBootstrap3 should not render those empty rows.

Don't blame Haskell for this, it is a common problem in all frameworks, they work fine for generic problems, but once you have a particular case, you need to be hacky, however in Yesod that is not the case, since you have the monadic forms and equality.

**Comment by felixSchl** [comment]
@ricksanchez Yes you are right, thank you for that comment. It helped me finally let go of the idea of the `render*` family of functions and instead just render directly to a widget from `MForm`. I think there's room for something similar to `AForm` that describes the structure (layout) of the form, as well as the resulting data structure. I've had another issue with using `AForm`s today which I [posted on stack overflow](https://stackoverflow.com/questions/46780837/map-over-aform-to-change-form-name-settings) that also leads me to believe that `AForm`s are a dead end for but the simplest use cases. Anyway... getting off topic.
